File: 226.invertTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:

    void invert(TreeNode* root) {
        if (root == NULL) return;
        TreeNode* left = root->left;
        TreeNode* right = root->right;

        root->left = right;
        root->right = left;

        invert(root->left);
        invert(root->right);
    }

    TreeNode* invertTree(TreeNode* root) {
        invert(root);
        return root;
    }
};

File: 226.invertTree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func invertTree(root *TreeNode) *TreeNode {
    if root == nil {
        return nil
    }

    tmp := invertTree(root.Right)
    root.Right = invertTree(root.Left)
    root.Left = tmp

    return root
}